primitives/core/src/error.rs

```diff
@@ -30,13 +30,15 @@ pub type ErrorString = BoundedVec<u8, MaxStringLength>;
 pub enum ErrorDetail {
 	// error when importing the parentchain blocks and executing indirect calls
 	ImportError,
+	// the direct or indirect request comes from an unauthorised sender
+	UnauthorisedSender,
 	// generic error when executing STF, the `ErrorString` should indicate the actual reason
 	StfError(ErrorString),
-	// error when sending stf request to the receiver
+	// error when sending stf request to the receiver fails
 	SendStfRequestFailed,
-	// error when challenge code not found
+	// error when challenge code can not found
```

Comment:
If we're going there :)

```suggestion
	// error when challenge code can not be found
```

Reply:
Thanks, fixed!